src/GraphQL/Resolvers/EventStreamResolver.cs

```diff
@@ -1,38 +1,40 @@
-using GraphQL.Subscription;
-
 namespace GraphQL.Resolvers
 {
-    public class EventStreamResolver<T> : IEventStreamResolver<T>
+    public class EventStreamResolver<T> : IEventStreamResolver
     {
-        private readonly Func<IResolveEventStreamContext, IObservable<T?>> _subscriber;
+        private readonly Func<IResolveFieldContext, ValueTask<IObservable<object?>>> _subscriber;
 
-        public EventStreamResolver(
-            Func<IResolveEventStreamContext, IObservable<T?>> subscriber)
+        public EventStreamResolver(Func<IResolveFieldContext, IObservable<T?>> subscriber)
         {
-            _subscriber = subscriber ?? throw new ArgumentNullException(nameof(subscriber));
-        }
+            if (subscriber == null)
+                throw new ArgumentNullException(nameof(subscriber));
 
-        public ValueTask<IObservable<T?>> SubscribeAsync(IResolveEventStreamContext context)
-            => new ValueTask<IObservable<T?>>(_subscriber(context));
+            if (typeof(T).IsValueType)
```

Comment:
where T : class

Reply:
I had put those in, which works great here, but it breaks compatibility with the field builders since there is no `where T : class` on the field builders.  So I did this instead.

Reply:
Perhaps you'd like to download the branch and see if you have any other ideas ?

Reply:
This is probably obvious, but it is necessary because `IObservable<int>` cannot be cast to `IObservable<object>`

Reply:
Of course I could use reflection to call the constructor and move the check into the field builder.

Reply:
Let's keep as is for now.

Reply:
- #2995